samples/readingEvents.go

```diff
@@ -0,0 +1,246 @@
+package samples
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/EventStore/EventStore-Client-Go/client"
+	"github.com/EventStore/EventStore-Client-Go/errors"
+	"github.com/EventStore/EventStore-Client-Go/options"
+	"github.com/EventStore/EventStore-Client-Go/stream_position"
+)
+
+func ReadFromStream(db *client.Client) {
+	// region read-from-stream
+	ropts := options.ReadStreamEventsOptionsDefault().
+		Position(stream_position.Start()).
+		Forwards()
+
+	stream, err := db.ReadStreamEvents(context.Background(), "some-stream", &ropts, 100)
+
+	if err != nil {
+		panic(err)
+	}
+
+	defer stream.Close()
+	// endregion read-from-stream
+	// region iterate-stream
+	for {
+		event, err := stream.Recv()
+
+		if err == io.EOF {
+			break
+		}
+
+		if err != nil {
+			panic(err)
+		}
+
+		fmt.Printf("Event> %v", event)
+	}
+	// endregion iterate-stream
+}
+
+func ReadFromStreamPosition(db *client.Client) {
+	// region read-from-position
+	ropts := options.ReadStreamEventsOptionsDefault().
+		Position(stream_position.Revision(10))
+
+	stream, err := db.ReadStreamEvents(context.Background(), "some-stream", &ropts, 20)
+
+	if err != nil {
+		panic(err)
+	}
+
+	defer stream.Close()
+	// endregion read-from-position
+	// region iterate-stream
+	for {
+		event, err := stream.Recv()
+
+		if err == io.EOF {
+			break
+		}
+
+		if err != nil {
+			panic(err)
+		}
+
+		fmt.Printf("Event> %v", event)
+	}
+	// endregion iterate-stream
+}
+
+func ReadFromStreamPositionCheck(db *client.Client) {
+	// region checking-for-stream-presence
+	ropts := options.ReadStreamEventsOptionsDefault().
+		Position(stream_position.Revision(10))
+
+	stream, err := db.ReadStreamEvents(context.Background(), "some-stream", &ropts, 100)
+
+	if err != nil {
+		panic(err)
+	}
+
+	defer stream.Close()
+
+	for {
+		event, err := stream.Recv()
+
+		if err == errors.ErrStreamNotFound {
```

Comment:
Why it's checked just in that example? Is this example special in any way? :thinking:

Reply:
It's a sample code to show a specific use case (like we do for other clients). In this case, it's to show how one can test if a stream exists.

Reply:
Yes yes, but we are checking it only in that example :thinking: 

I'm also thinking if
1. is it handled properly? should we not break the loop? :thinking: 
2. can't this error be returned from `db.ReadStreamEvents(context.Background(), "some-stream", &ropts, 100)` ? or it's expected behaviour that this loop will continue until stream will be created?

Reply:
No, it's not something you should check for every example.

about your second point, you are right, I'll make the change

Reply:
by making the change I meant to return the not-found error from `db.ReadStreamEvents`